processing/src/main/java/org/apache/druid/query/groupby/having/GreaterThanHavingSpec.java

```diff
@@ -119,4 +122,15 @@ public String toString()
     sb.append('}');
     return sb.toString();
   }
+
+  @Override
+  public byte[] getCacheKey()
+  {
+    return new CacheKeyBuilder(HavingSpecUtil.CACHE_TYPE_ID_GREATER_THAN)
+        .appendString(aggregationName)
+        .appendByteArray(StringUtils.toUtf8(String.valueOf(value)))
+        .appendStrings(aggregators.keySet())
+        .appendCacheables(aggregators.values())
```

Comment:
not sure the best way to add the `aggregators` map to CacheKeyBuilder, I am adding the keys and values separately, is that okay ?

Reply:
You don't need to add them to the cache key -- only the `aggregation` and `value` matter. The idea is that anything that changes the meaning or effect of the operator should be part of the cache key, but not other stuff. `aggregators` is just some extra information that the HavingSpec uses to do its job.